src/lamplib/src/genny/tasks/run_tests.py

```diff
@@ -201,82 +227,63 @@ def _setup_resmoke(
             check=True,
             capture=False,
         )
-        cmd_runner.run_command(
-            # If changing this sha, you may need to use later binaries
-            # in the _canned_artifacts dict.
-            cmd=["git", "checkout", "cf8ddbdf99ae7fb2d96ca9f350224ffa193e40b3"],
-            cwd=mongo_repo_path,
-            check=True,
-            capture=False,
-        )
-    else:
-        SLOG.info("Using existing mongo repo checkout", mongo_repo_path=mongo_repo_path)
-        cmd_runner.run_command(
-            cmd=["git", "rev-parse", "HEAD"], check=False, cwd=mongo_repo_path, capture=False,
-        )
+        checkout_required = True
```

Comment:
Why do we checkout if using the existing repo? Especially since a comment further down says "only required if the local repo didn't exist and had to be cloned"

In particular, this could be the user-specified repo, right? In that case I think we agreed we wouldn't touch it?

I do think we wanna check if this is the repo at the default location or the user-specified one, and only checkout with the former (so that genny updates to the checked-out repo are applied)

(Sorry for being so overfocused on that! I made a mistake in this file that deleted someone's local mongo repo and want to avoid a similar issue.)